### ISSUE TITLE
Replacing layers in a trained net

### ISSUE THREAD

**Issue by akosiorek** [BVLC/caffe#328]
I'd like to replace fc7 (4096 outputs) and fc8 (1000 outputs) layers from the imagenet classifier for custom fc layers with 2048 and 14 outputs respectively.

I had my dataset fed through the imagenet classifier up to the fc6 layer, extracted outputs and stored them as Datums in a leveldb. Then, I trained a tiny neural network with a data layer, 2 inner product and dropout layers, a softmax and a loss layer.

Finally, I prepared a prototxt network model, initialized a Net with it and copied imagnet classifier's layers up to the fc6 layer. Then I added all layers from my new neural network (without the data layer) on top of that.

The problem is: results I get are nowhere near  the results of the small neural network fed with buffered data. What could go wrong? Any suggestions? How would you go about merging two networks?
